File: Thesis_data_scripts/executions/fib.js
function main(params) {
  let n = params.n
  if (n <= 0) {
    return { result: null };
  } else if (n === 1) {
    return { result: 0 };
  } else if (n === 2) {
    return { result: 1 };
  } else {
    return { result: fibonacciLastNumberRec(n) };
  }
}

function fibonacciLastNumberRec(n) {
  if (n <= 0) {
    return 0;
  } else if (n === 1) {
    return 1;
  } else {
    return fibonacciLastNumberRec(n - 1) + fibonacciLastNumberRec(n - 2);
  }
}

File: Thesis_data_scripts/executions/fib.py
def main(n):
    if n <= 0:
        return {'result': None}
    elif n == 1:
        return {'result': 0}
    elif n == 2:
        return {'result': 1}
    else:
        a, b = 0, 1
        for _ in range(2, n):
            a, b = b, a + b
        return {'result': b}



